mvc/src/main/java/nextstep/mvc/controller/tobe/AnnotationHandlerMapping.java

```diff
@@ -22,9 +31,51 @@ public AnnotationHandlerMapping(final Object... basePackage) {
 
     public void initialize() {
         log.info("Initialized AnnotationHandlerMapping!");
+
+        flatAllMethodsFromControllers()
+                .forEach(this::appendExecutionPerMethod);
+    }
+
+    private List<Method> flatAllMethodsFromControllers() {
+        final Reflections reflections = new Reflections(basePackage);
+
+        return reflections.getTypesAnnotatedWith(Controller.class)
+                .stream()
+                .map(Class::getMethods)
+                .flatMap(Arrays::stream)
+                .collect(Collectors.toUnmodifiableList());
```

Comment:
filter를 추가해준다면 50번째 라인의 if문을 제거할 수 있을 듯합니다!
```suggestion
        reflections.getTypesAnnotatedWith(Controller.class)
                .stream()
                .map(Class::getMethods)
                .flatMap(Arrays::stream)
                .filter(method -> method.isAnnotationPresent(RequestMapping.class))
                .collect(Collectors.toUnmodifiableList());
```

Reply:
```java
    private List<Method> flatAllMethodsFromControllers() {
        final Reflections reflections = new Reflections(basePackage);

        return reflections.getTypesAnnotatedWith(Controller.class)
                .stream()
                .map(Class::getMethods)
                .flatMap(Arrays::stream)
                .filter(method -> method.isAnnotationPresent(RequestMapping.class))
                .collect(Collectors.toUnmodifiableList());
    }

    private void appendExecutionPerMethod(final Method method) {
        final RequestMapping requestMapping = method.getDeclaredAnnotation(RequestMapping.class);

        final HandlerExecution handlerExecution = new HandlerExecution(method);
        appendExecutionPerRequests(requestMapping, handlerExecution);
    }
```
말씀하신대로 `appendExecutionPerMethod`의 if문을 제거하고자 `flatAllMethodsFromControllers`의 stream filter로 해당 로직을 옮기게 되면, `appendExecutionPerMethod`의 `method.getDeclaredAnnotation(RequestMapping.class)` 로직에 대해 안정성을 보장할 수 없다고 생각하는데, 케이는 그럼에도 로직을 옮기는 것이 옳다고 판단하시나요??

Reply:
filter로 AnnotationPresent이지 않은 method를 걸러준 후에 appendExecutionPerMethod 메서드를 실행하기 때문에 현재 코드랑 같은 기능을 한다고 생각해서 코멘트 드렸는데, 혹시 어떤 부분이 다른지 추가설명 부탁드려도 될까요?

Reply:
현재 코드의 흐름상 문제가 없지만, 분리된 메서드는 그 자체로 독립성을 보장받을 수 있어야 한다고 생각해요. 그래야만 재사용성을 확보할 수 있으니까요. 그런데 `appendExecutionPerMethod` 메서드에서 `isAnnotationPresent` 조건 로직을 제거하게 된다면, `getDeclaredAnnotation`을 호출할 때 예외가 발생할 수 있기 때문에 안정성을 보장할 수 없다고 생각해요!!